src/training/save_states.rs

```diff
@@ -228,19 +233,95 @@ pub unsafe fn save_states(module_accessor: &mut app::BattleObjectModuleAccessor)
             set_damage(module_accessor, save_state.percent);
         }
 
+        // if the fighter is Popo, change the state to one where only Nana can move
+        // This is needed because for some reason, outside of frame by frame mode,
+        // Popo will keep trying to move instead of letting Nana move if you just
+        // change the state back to PosMove
+        let prev_status_kind = StatusModule::prev_status_kind(module_accessor, 0);
+        if prev_status_kind == FIGHTER_STATUS_KIND_REBIRTH && fighter_is_popo {
+            save_state.state = NanaPosMove;
```

Comment:
Why create another state,  can't we reuse PosMove and add the new Nana specific conditions there?

Reply:
That's exactly what I thought as well. The issue was that when I tried to just reuse PosMove, Popo would always be the only fighter who would check that condition, except during frame by frame mode. I do not understand why this is the case, and I tried implementing a few different ways to make sure Nana would use PosMove outside of frame by frame, but didn't have any luck. Adding the other state worked perfectly nearly the instant I tried it, so I thought I'd roll with it. If you have any ideas as to why Nana would only check that during frame by frame and how to fix it we could give it a shot, but I don't believe I have the knowledge to figure out why that happens at this point.

Reply:
Hm, I guess I see the need for another state like NanaPosMove-- it must be because the save_state.state is shared between the two ICs, and perhaps Popo steals it each frame before Nana can move before it becomes NoAction. Still, can you at least have the PosMove block check for state == PosMove || state == NanaPosMove? That way we don't have a lot of duplicated code. You'll just need another check for that if iys NanaPosMove, we also move to NoAction or something equivalent.

Reply:
You're exactly right that the save_state.state is shared between the two ICs. It also did look to me like Popo gets to it first before Nana does. The confusing part to me was that there were times when Popo would check PosMove multiple times in one frame, and Nana would not ever check it (and this didn't happen in frame by frame). That solution sounds pretty good though, I'll clean up what I'm working on right now and give it a roll.